src/aatoolbox/utils/io.py

```diff
@@ -82,35 +83,48 @@ def parse_yaml(filename: Union[str, Path]) -> dict:
     return config
 
 
-def check_file_existence(func: F) -> F:
+@wrapt.decorator
+def check_file_existence(wrapped: F, instance, args, kwargs) -> F:
     """
     Don't overwrite existing data.
 
     Avoid recreating data if it already exists and if clobber not
-    toggled by user. Can be used to wrap a function whose first parameter
-    is a filepath.
+    toggled by user. Used to wrap functions that accept filepath
+    as a keyword argument.
 
     Parameters
     ----------
-    func : function
-        The function to wrap. The first parameter of this function must
-        be the filepath (of type Path), and it can also have an optional
+    wrapped : function
+        The function to wrap. The function must have "filepath" as
+        a keyword parameter, and it can also have an optional
         "clobber" boolean keyword parameter.
+    instance :
+        Object the wrapped function is bound to. Not used within, but
+        ensures that instance methods do not pass `self` to args.
+    args :
+        List of positional arguments.
+    kwargs :
+        Dictionary of keyword arguments
 
     Returns
     -------
     If filepath exists, returns filepath. Otherwise, returns the result of
     the decorated function.
-
     """
-
-    def wrapper(filepath: Path, *args, **kwargs):
-        if filepath.exists() and not kwargs.get("clobber", False):
-            logger.info(
-                f"File {filepath} exists and clobber set to False, "
-                f"using existing files"
+    filepath = kwargs.get("filepath")
+    if filepath is None:
+        raise KeyError(
+            (
+                "`filepath` must be passed as a keyword "
+                "argument for the `check_file_existence`"
+                " decorator to work."
             )
-            return filepath
-        return func(filepath, *args, **kwargs)
-
-    return cast(F, wrapper)
+        )
+    if filepath.exists() and not kwargs.get("clobber", False):
```

Comment:
I think this part here is not covered by any tests, would it be possible to add some? Sorry Seth, I know I'm terrible because I'm the one who wrote the function in the first place and didn't make tests! Only if you have time.

Reply:
Will add in as hopefully final step!